samza-kafka/src/main/scala/org/apache/samza/checkpoint/kafka/KafkaCheckpointManager.scala

```diff
@@ -53,6 +53,8 @@ class KafkaCheckpointManager(checkpointSpec: KafkaStreamSpec,
                              checkpointMsgSerde: Serde[Checkpoint] = new CheckpointSerde,
                              checkpointKeySerde: Serde[KafkaCheckpointLogKey] = new KafkaCheckpointLogKeySerde) extends CheckpointManager with Logging {
 
+  var MAX_RETRIES_ON_FAILURE = 50
```

Comment:
Make this a val. Move to line 152, where the retry loop is.

Reply:
I want to override retries in test(set to 1)(since it's relevant only to one api, i don't think it's logical to inject it through the constructor).

Reply:
It's fine to add it to the constructor. That's what they're for - setting up the class' state.

If you really want to keep this as a field, Scala doesn't use the ALL_CAPS convention for naming constants. Change to AllCaps instead.

Reply:
Changed to AllCaps.